### ISSUE TITLE
Invalid value in the cpu_usage column from gp_toolkit.gp_resgroup_status

### ISSUE THREAD

**Issue by andr-sokolov** [greenplum-db/gpdb#14154]
## Bug Report

Affected branches: master and 6X_STABLE. 

The contents of the cpu_usage field from gp_toolkit.gp_rec group_status depends on query type. In the case of a simple select, everything is ok. In the case of insert-select, the field contains information about master only. This leads to an error at the moment of analyzing the table, as the contents of the field do not correspond to json format.
```
postgres=# select cpu_usage from gp_toolkit.gp_resgroup_status;
                 cpu_usage                 
-------------------------------------------
 {"-1":0.00, "0":0.00, "1":0.00, "2":0.00}
 {"-1":0.21, "0":0.08, "1":0.08, "2":0.08}
(2 rows)

postgres=# create table tst_json (cpu_usage json null);
NOTICE:  Table doesn't have 'DISTRIBUTED BY' clause, and no column type is suitable for a distribution key. Creating a NULL policy entry.
CREATE TABLE
postgres=# insert into tst_json select cpu_usage from gp_toolkit.gp_resgroup_status;
INSERT 0 2
postgres=# table tst_json;
 cpu_usage 
-----------
 "-1":0.00
 "-1":0.03
(2 rows)

postgres=# analyze tst_json;
ERROR:  invalid input syntax for type json
DETAIL:  Expected end of input, but found ":".
CONTEXT:  JSON data, line 1: "-1":...
postgres=# 
```

If the analysis is performed automatically at the moment of insertion, the error appears as follows:
```
postgres=# insert into tst_json select cpu_usage from gp_toolkit.gp_resgroup_status;
ERROR:  invalid input syntax for type json
DETAIL:  Expected end of input, but found ":".
CONTEXT:  JSON data, line 1: "-1":...
postgres=# 
```

The gp_toolkit.gp_resgroup_status view is based on the pg_resgroup_get_status function. The reason for the error is that the function is called on entry db only in the case of insert-select. It should be called on all segments like in the case of a simple select.
A similar problem occurs when calling the pg_resgroup_get_status_kv function.

The error can be fixed by setting the proexeclocation property of the function to 'i' (PROEXECLOCATION_INITPLAN). But here (https://github.com/greenplum-db/gpdb/pull/11435) I found "Changing catalog for stable release is something we would like to avoid unless absolutely necessary for critical issues."

Is there any other way to fix the error, other than changing the system catalog?

### OS version and uname -a
```
$ cat /etc/os-release
NAME="Simply Linux"
VERSION="10.1 (Captain Finn)"
ID=altlinux
VERSION_ID=10.1
PRETTY_NAME="Simply Linux 10.1 (Captain Finn)"
ANSI_COLOR="1;36"
CPE_NAME="cpe:/o:alt:slinux:10.1"
BUILD_ID="Simply Linux 10.0"
HOME_URL="https://www.basealt.ru/products/simplylinux/"
BUG_REPORT_URL="https://bugs.altlinux.org/"
$ uname -a
Linux sl10 5.10.117-std-def-alt1 #1 SMP Wed May 18 12:57:04 UTC 2022 x86_64 GNU/Linux
```

### autoconf options used ( config.status --config )
```
$ ./config.status --config
'--enable-debug-extensions' '--with-python' '--disable-gpcloud' '--enable-depend' '--enable-cassert' 'CFLAGS=-O0 -g3 -ggdb -fno-omit-frame-pointer -Wno-error=deprecated-copy' 'CXXFLAGS=-O0 -g3 -fno-omit-frame-pointer -Wno-error=deprecated-copy' '--prefix=/home/keremet/compile/gpdb_bin'
```
### Installation information ( pg_config )

```
$ pg_config
BINDIR = /home/keremet/compile/gpdb_bin/bin
DOCDIR = /home/keremet/compile/gpdb_bin/share/doc/postgresql
HTMLDIR = /home/keremet/compile/gpdb_bin/share/doc/postgresql
INCLUDEDIR = /home/keremet/compile/gpdb_bin/include
PKGINCLUDEDIR = /home/keremet/compile/gpdb_bin/include/postgresql
INCLUDEDIR-SERVER = /home/keremet/compile/gpdb_bin/include/postgresql/server
LIBDIR = /home/keremet/compile/gpdb_bin/lib
PKGLIBDIR = /home/keremet/compile/gpdb_bin/lib/postgresql
LOCALEDIR = /home/keremet/compile/gpdb_bin/share/locale
MANDIR = /home/keremet/compile/gpdb_bin/share/man
SHAREDIR = /home/keremet/compile/gpdb_bin/share/postgresql
SYSCONFDIR = /home/keremet/compile/gpdb_bin/etc/postgresql
PGXS = /home/keremet/compile/gpdb_bin/lib/postgresql/pgxs/src/makefiles/pgxs.mk
CONFIGURE = '--enable-debug-extensions' '--with-python' '--disable-gpcloud' '--enable-depend' '--enable-cassert' 'CFLAGS=-O0 -g3 -ggdb -fno-omit-frame-pointer -Wno-error=deprecated-copy' 'CXXFLAGS=-O0 -g3 -fno-omit-frame-pointer -Wno-error=deprecated-copy' '--prefix=/home/keremet/compile/gpdb_bin'
CC = gcc
CPPFLAGS = -D_GNU_SOURCE -I/home/keremet/compile/gpdb_bin/include
CFLAGS = -Wall -Wmissing-prototypes -Wpointer-arith -Wendif-labels -Wmissing-format-attribute -Wformat-security -fno-strict-aliasing -fwrapv -fexcess-precision=standard -fno-aggressive-loop-optimizations -Wno-unused-but-set-variable -Wno-address -Werror=implicit-fallthrough=3 -Wno-format-truncation -Wno-stringop-truncation -O0 -g3 -ggdb -fno-omit-frame-pointer -Wno-error=deprecated-copy -std=gnu99 -Werror=uninitialized -Werror=implicit-function-declaration -I/home/keremet/compile/gpdb_bin/include
CFLAGS_SL = -fPIC
LDFLAGS = -Wl,--as-needed -Wl,-rpath,'/home/keremet/compile/gpdb_bin/lib',--enable-new-dtags -L/home/keremet/compile/gpdb_bin/lib
LDFLAGS_EX = 
LDFLAGS_SL = 
LIBS = -lpgcommon -lpgport -lxerces-c -lrt -lzstd -lrt -lcrypt -ldl -lm -L/home/keremet/compile/gpdb_bin/lib
VERSION = PostgreSQL 9.4.26
```

**Comment by kainwen** [comment]
Hi, thanks for the reporting.

My idea (for 6x):

1. disallow the function execute on entryDB (like the same idea of https://github.com/greenplum-db/gpdb/pull/13075
2. use this skill as workaround for your SQL:

```sql
gpadmin@test:/sys/fs/cgroup$ psql
psql (9.4.26)
Type "help" for help.

gpadmin=#  --explain insert into tst_json select cpu_usage from gp_toolkit.gp_resgroup_status;
gpadmin=# \e
                                           QUERY PLAN
------------------------------------------------------------------------------------------------
 Insert on tst_json  (cost=14.90..15.90 rows=50 width=32)
   InitPlan 1 (returns $0)  (slice2)
     ->  Hash Join  (cost=1.05..14.90 rows=10 width=32)
           Hash Cond: (s.groupid = r.oid)
           ->  Function Scan on pg_resgroup_get_status s  (cost=0.00..10.00 rows=1000 width=36)
           ->  Hash  (cost=1.02..1.02 rows=1 width=4)
                 ->  Seq Scan on pg_resgroup r  (cost=0.00..1.02 rows=2 width=4)
   ->  Redistribute Motion 1:2  (slice1; segments: 1)  (cost=0.00..1.00 rows=100 width=32)
         ->  Function Scan on unnest  (cost=0.00..1.00 rows=50 width=32)
 Optimizer: Postgres query optimizer
(10 rows)

gpadmin=#

insert into tst_json select * from unnest(array(select cpu_usage from gp_toolkit.gp_resgroup_status));
INSERT 0 2
gpadmin=#
```

**Comment by SmartKeyerror** [comment]
Fixed by #14959, so close this issue.
